### PR TITLE
flix: init at 0.35.0

### PR DESCRIPTION
**Pull request by athas**
###### Description of changes

Added derivation for the [Flix](https://github.com/flix/flix) compiler, a somewhat popular project (1700 stars on GitHub).

Note that this derivation uses the upstream .jar file, as seems relatively common for Java-based derivations.  I suppose I can change the derivation to use `sbt` to build from source, if that is desirable, but I don't think I know of any existing derivations that use `sbt`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by fabianhjr — APPROVED** [review]
```
1 package built:
flix

$ /nix/store/hbr5j4sd0rrdlh9hybzsiglvnf2j80la-nix-2.13.3/bin/nix-shell /home/fabian/.cache/nixpkgs-review/pr-227034/shell.nix

[nix-shell:~/.cache/nixpkgs-review/pr-227034]$ flix
No `flix.toml'. Will load source files from `*.flix`, `src/**`, and `test/**`.
     __   _   _
    / _| | | (_)            Welcome to Flix 0.35.0
   | |_  | |  _  __  __
   |  _| | | | | \ \/ /     Enter an expression to have it evaluated.
   | |   | | | |  >  <      Type ':help' for more information.
   |_|   |_| |_| /_/\_\     Type ':quit' or press 'ctrl + d' to exit.
      
flix> 10 + 1

11    
```

LGTM

File: pkgs/development/compilers/flix/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  dontUnpack = true;
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
+    runHook preInstall
+    export JAR=$out/share/java/${pname}/${pname}.jar
+    install -D $src $JAR
+    makeWrapper ${jre}/bin/java $out/bin/${pname} \
+      --add-flags "-jar $JAR"
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/flix/default.nix:21]:
```suggestion
    runHook preInstall

    export JAR=$out/share/java/flix/flix.jar
    install -D $src $JAR
    makeWrapper ${jre}/bin/java $out/bin/flix \
      --add-flags "-jar $JAR"

```

File: pkgs/development/compilers/flix/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "The Flix Programming Language";
+    homepage = "https://github.com/flix/flix";
+    sourceProvenance = with sourceTypes; [ binaryBytecode ];
+    license = licenses.asl20;
+    maintainers = with maintainers; [ athas ];
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/flix/default.nix:31]:
```suggestion
    inherit (jre.meta) platforms;
```

**Comment by athas** [comment]
Anything else that remains to be done?

**Comment by fabianhjr** [comment]
Don't think so, probably best to ping another nixpkgs-commiter. Going to send a message to the matrix dev channel.

**Pull request merged**